tests/instances.tcl

```diff
@@ -281,7 +281,7 @@ proc parse_options {} {
             dict set ::global_config $val $val2
             incr j 2
         } elseif {$opt eq "--help"} {
-            puts "--single <pattern>      Only runs tests specified by pattern."
+            puts "--single <pattern>      Only runs tests specified by pattern. Regexp must start with '/'."
```

Comment:
i now realize i mixed two things (for some reason i thought instances.tcl supports `--only`).
but in fact it used `search_pattern_list` for it's `--single`, which unlike the test_helper's `--single`, was using a sub-string.
so in order not to break that, we must either let it keep doing sub-string matching (no need for regex or any fancy things).
or maybe we can let it use regex (since by default it's the same as sub-string), without requiring the `/` prefix.
i.e. because the tests are named `23-multiple-slot-operations.tcl` and alike, maybe it doesn't matter much if it's a regex or a substring.

but anyway, since we need to give up the requirement for `/` prefix, we can't use the generic mechanism, and i think we should just let it search for sub-string via a dedicated function, and not mix it with the other (`--only`, `--skiptest`, and `--skipfile`) which are all about test names and not unit files.

Reply:
fixed in https://github.com/redis/redis/pull/10741/commits/032cc08f69cf70a5a2394608971ef792b70e93e4